src/main/java/org/gridsuite/ds/server/service/DynamicSimulationWorkerService.java

```diff
@@ -220,14 +225,18 @@ public CompletableFuture<DynamicSimulationResult> getCompletableFuture(DynamicSi
                 runContext.getNetworkUuid(), parameters.getStartTime(), parameters.getStopTime());
 
         DynamicSimulation.Runner runner = DynamicSimulation.find(provider);
-        return runner.runAsync(runContext.getNetwork(),
+        CompletableFuture<DynamicSimulationResult> dynamicSimulationResultCompletableFuture = runner.runAsync(runContext.getNetwork(),
                 dynamicModelsSupplier,
                 eventModelsSupplier,
                 outputVariablesSupplier,
                 runContext.getVariantId() != null ? runContext.getVariantId() : VariantManagerConstants.INITIAL_VARIANT_ID,
                 getComputationManager(),
                 parameters,
                 runContext.getReportNode());
+        return dynamicSimulationResultCompletableFuture.thenCompose(result -> CompletableFuture.supplyAsync(() -> {
+            // throw new RuntimeException("Simulation failed");
```

Comment:
Return or throw ? :)

Reply:
This is code for testing exception.. it will be removed